### PR TITLE
Add scalar tensor system and tags

### PR DESCRIPTION
**Pull request by guilara**
## Proposed changes

Add new namespace  and tags/aliases for the combined system of CurvedScalarWave and generalized harmonic.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/ScalarTensor/System.hpp
Comment (nilsvu) [inline src/Evolution/Systems/ScalarTensor/System.hpp:16]: Please add some docs here :) Add the equations and/or link to the GH and CSW systems.

File: src/Evolution/Systems/ScalarTensor/Tags.hpp
Comment (nilsvu) [inline src/Evolution/Systems/ScalarTensor/Tags.hpp:15]: Add docstring so this shows up in the dox

File: src/Evolution/Systems/ScalarTensor/Tags.hpp
Comment (nilsvu) [inline src/Evolution/Systems/ScalarTensor/Tags.hpp:22]: Please add some docs in the `ScalarTensor` namespace to clarify where you need this

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: src/Evolution/Systems/ScalarTensor/System.hpp
Comment (guilara) [reply src/Evolution/Systems/ScalarTensor/System.hpp:16]: Added an expanded description below.

File: src/Evolution/Systems/ScalarTensor/Tags.hpp
Comment (nilsvu) [inline src/Evolution/Systems/ScalarTensor/Tags.hpp:22]: Please template on `typename DataType, size_t Dim, typename Fr = Frame::Inertial>` for consistency with other tensor tags.

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: src/Evolution/Systems/ScalarTensor/Tags.hpp
Comment (guilara) [reply src/Evolution/Systems/ScalarTensor/Tags.hpp:22]: Added template

File: src/Evolution/Systems/ScalarTensor/Tags.hpp
Comment (nilsvu) [inline src/Evolution/Systems/ScalarTensor/Tags.hpp]: `tnsr::aa<DataType, Dim, Fr>`

**Review by nilsvu — CHANGES_REQUESTED** [review]
Looks good, please squash this in.

File: src/Evolution/Systems/ScalarTensor/Tags.hpp
Comment (guilara) [reply src/Evolution/Systems/ScalarTensor/Tags.hpp]: Thanks Nils, I've made the change

**Review by nilsvu — DISMISSED** [review]
Looks good. CI has a few complaints. Also, do you want to wait for the other reviews that you requested?

**Review by nilsdeppe — COMMENTED** [review]
LGTM, I don't need to look at it again for it to get merged :)

**Comment by guilara** [comment]
I updated the test. Hopefully it won't complain now.

I added @nikwit to make him aware of this PR. I think it his input will be useful later on.

**Comment by nilsvu** [comment]
Just one CI complaint left

**Comment by guilara** [comment]
I think the last complaint was due to cmake looking for the .hpp files in the CMakeLists.txt file. Previously I had only written the license in it as I didn't want to add those details now. In the end I've added more details to that file. I also added a couple of mostly empty files (StressEnergy.hpp/cpp) to pass at least one 'source' file to cmake and avoid more complaints.

**Review by nilsvu — APPROVED** [review]
The empty files aren't necessary (CMake would be fine without a `sources` list for example), but don't hurt either.

Want to wait for @nikwit before merging?

**Comment by guilara** [comment]
I would say not to wait as this is just tags. But I'll ask for his review for the following

**Pull request merged**
